From 96c897d3a6cb76928d7981f60f832afba1bcc01b Mon Sep 17 00:00:00 2001
From: DuanqiangWen <duanqiangwen@net-swift.com>
Date: Wed, 9 Nov 2022 02:23:52 -0500
Subject: [PATCH] openeuler: configs: delete txgbe/Kconfig, add txgbe_config to
 netswift/Kconfig
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 96c897d3a6cb76928d7981f60f832afba1bcc01b
Modified-by-SEL: Yes, modified due to different context


driver inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5VU60
CVE: NA

add CONFIG_NETSWIFT_NGBE=m in arm64 openeuler_defconfig
add CONFIG_TXGBE_HWMON for hardware monitoring support
add CONFIG_TXGBE_DEBUG_FS for debug_fs support
add CONFIG_TXGBE_POLL_LINK_STATUS for polling
to get phy status support instead of interrupt.
add CONFIG_TXGBE_SYSFS for sysfs support

Signed-off-by: DuanqiangWen <duanqiangwen@net-swift.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/netswift/Kconfig       |   55 +++++++++++++++++++++++++++-
 drivers/net/ethernet/netswift/txgbe/Kconfig |   13 ------
 2 files changed, 54 insertions(+), 14 deletions(-)
 delete mode 100644 drivers/net/ethernet/netswift/txgbe/Kconfig

--- a/drivers/net/ethernet/netswift/Kconfig
+++ b/drivers/net/ethernet/netswift/Kconfig
@@ -16,7 +16,6 @@ config NET_VENDOR_NETSWIFT
 
 if NET_VENDOR_NETSWIFT
 
-source "drivers/net/ethernet/netswift/txgbe/Kconfig"
 config NGBE
 	tristate "Netswift PCI-Express Gigabit Ethernet support"
 	depends on PCI
@@ -73,4 +72,58 @@ config NGBE_SYSFS
 
 	  If unsure, say N.
 
+config TXGBE
+        tristate "Netswift PCI-Express 10Gigabit Ethernet support"
+        depends on PCI
+        imply PTP_1588_CLOCK
+        help
+          This driver supports Netswift 10gigabit ethernet adapters.
+ 	  For more information on how to identify your adapter, go
+          to <http://www.net-swift.com>
+
+          To compile this driver as a module, choose M here. The module
+	  will be called txgbe.
+
+config TXGBE_HWMON
+	bool "Netswift PCI-Express 10Gigabit adapters HWMON support"
+	default n
+	depends on TXGBE && HWMON && !(TXGBE=y && HWMON=m)
+	help
+	  Say Y if you want to expose thermal sensor data on these devices.
+	  For more information on how to use your adapter, go
+	  to <http://www.net-swift.com>
+
+	  If unsure, say N.
+
+config TXGBE_DEBUG_FS
+	bool "Netswift PCI-Express 10Gigabit adapters debugfs support"
+	default n
+	depends on TXGBE
+	help
+	  Say Y if you want to setup debugfs for these devices.
+	  For more information on how to use your adapter, go
+	  to <http://www.net-swift.com>
+
+	  If unsure, say N.
+
+config TXGBE_POLL_LINK_STATUS
+	bool "Netswift PCI-Express 10Gigabit adapters poll mode support"
+	default n
+	depends on TXGBE
+	help
+	  Say Y if you want to turn these devices to poll mode instead of interrupt-trigged TX/RX.
+	  For more information on how to use your adapter, go
+	  to <http://www.net-swift.com>
+
+	  If unsure, say N.
+config TXGBE_SYSFS
+	bool "Netswift PCI-Express 10Gigabit adapters sysfs support"
+	default n
+	depends on TXGBE
+	help
+	  Say Y if you want to setup sysfs for these devices.
+	  For more information on how to use your adapter, go
+	  to <http://www.net-swift.com>
+
+	  If unsure, say N.
 endif # NET_VENDOR_NETSWIFT
--- a/drivers/net/ethernet/netswift/txgbe/Kconfig
+++ /dev/null
@@ -1,13 +0,0 @@
-#
-# Netswift driver configuration
-#
-
-config TXGBE
-	tristate "Netswift 10G Network Interface Card"
-	default n
-	depends on PCI_MSI && NUMA && PCI_IOV && DCB
-	help
-	  This driver supports Netswift 10G Ethernet cards.
-	  To compile this driver as part of the kernel, choose Y here.
-	  If unsure, choose N.
-	  The default is N.
